### PR TITLE
KAFKA-13553: Add PAPI Window and Session store tests for IQv2

### PR DESCRIPTION
**Pull request by vvcephei**
During some recent reviews, @mjsax pointed out that StateStore layers
are constructed differently the stores are added via the PAPI vs. the DSL.

This PR adds PAPI construction for Window and Session stores to the
IQv2StoreIntegrationTest so that we can ensure IQv2 works on every
possible state store.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java [hunk 1]
```diff
@@ -800,13 +914,13 @@ public void verifyStore() {
                     throw new AssertionError(queryResult.toString());
                 }
                 assertThat(partitionResult.getFailureReason(), is(FailureReason.UNKNOWN_QUERY_TYPE));
-                assertThat(partitionResult.getFailureMessage(), is(
+                assertThat(partitionResult.getFailureMessage(), matchesPattern(
                     "This store"
-                        + " (class org.apache.kafka.streams.state.internals.MeteredTimestampedWindowStore)"
+                        + " \\(class org.apache.kafka.streams.state.internals.Metered.*WindowStore\\)"
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java:918]: Regardless of whether you supply a bytes store to the DSL, you wind up wrapped in a Timestamped store. Now that we're registering the non-timestamped store directly with the PAPI, we can also get a non-timestamped metered store, hence swapping this for a regex that matches both `MeteredWindowStore` and `MeteredTimestampedWindowStore`.

**Review by vvcephei — COMMENTED** [review]
Hey, @guozhangwang , can you spare a quick look at this?

It's a follow-on to https://github.com/apache/kafka/pull/11624 to add window and session stores to the test (I left them off to keep the other PR short and focused).

File: streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java [hunk 2]
```diff
@@ -626,6 +627,118 @@ public void process(final Record<Integer, Integer> record) {
...
+        };
+        if (cache) {
+            sessionStoreStoreBuilder.withCachingEnabled();
+        } else {
+            sessionStoreStoreBuilder.withCachingDisabled();
+        }
+        if (log) {
+            sessionStoreStoreBuilder.withLoggingEnabled(Collections.emptyMap());
+        } else {
+            sessionStoreStoreBuilder.withCachingDisabled();
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java]: Seems you got the same :) and ditto elsewhere.

**Review by guozhangwang — APPROVED** [review]

File: streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java (hunk 2 above)
Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java]: Huh, it looks like I missed updating the PAPI method in the other PR, and then amplified it here when I copy/pasted it for the other stores. Sorry about that!

**Pull request merged**